spec/pages_spec.rb

```diff
@@ -15,9 +18,23 @@
 
     it 'has only valid internal hyperlinks' do
       page.all(:css, 'a').each do |link|
-        next if link.text == '' || link[:href].match(/(http|\/\/).*/)
-        page.find(:xpath, link.path).click
-        expect(page.status_code).to be(200), "expected link '#{link.text}' to work"
+        next if link.text.nil? || link.text.empty? || link[:href].match(/^http.*/) || link.path.nil? || urls.include?(link[:href])
+        begin
+          page.find(:xpath, link.path).click
+	rescue NoMethodError
+	  require 'irb';binding.irb
+	end
+        urls.push link.path
+	expect(page.status_code).to be(200), "expected link '#{link.text}' to work"
+        visit p
+      end
+    end
+
+    it 'has valid external hyperlinks' do
+      page.all(:css, 'a').each do |link|
+        next if link.text.nil? || link.text.empty? || !link[:href].match(/^http.*/) || urls.include?(link[:href])
+	urls.push(link[:href])
+	expect(url_exists? link[:href], status).to be_truthy, "expected link '#{link.text}' => '#{link[:href]}' to work (Error '#{status}')"
```

Comment:
It is probably not a good idea to nest this. Move `url_exists?` out of the `expect`.